src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -289,13 +289,12 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 
 					ApiResponse res;
 					switch (reqType) {
-					case action:	
-						// TODO Handle POST requests - need to read these in and then parse params from POST body
-						/*
-						if (getOptionsParamApi().isPostActions()) {
-							throw new ApiException(ApiException.Type.DISABLED);
+					case action:
+						if (requestHeader.getMethod().toUpperCase().equals("POST")) {
+							String body = httpIn.readRequestBody(requestHeader).toString();
```

Comment:
It should be checked that the Content-Type is the one that we expect, otherwise it should throw an ApiException (with type `UNSUPPORTED_REQUEST`?).